### DISCUSSION TITLE
How do I setup CRUD list operation when the table doesn't have an ID column?

### DISCUSSION THREAD

**Discussion by cvassios-eibach** [Laravel-Backpack/community-forum#1052] [Q&A (Help)] [answered]
It looks like it defaults to always ordering by the ID, but in this case, my table doesn't have an ID column so it throws an error:
`Column not found: 1054 Unknown column 'id' in 'order clause'`
```
SELECT
  *
FROM
  `product_release_product`
ORDER BY
  `product_release_id` DESC,
  `id` DESC
limit
  10
```

**Comment by karandatwani92** [comment]
If your table doesn't have `id` as the primary key, You should set one in your Model. Example:
```
protected $primaryKey = 'barcode';
public $incrementing = false;
protected $keyType = 'string';
```

**Reply by cvassios-eibach** [reply]
My table is a n-n pivot table, so I'm not using a primary key. Is there a way to make it work with a composite primary key?

**Reply by karandatwani92** [reply, marked as answer]
"Composite" primary keys are [not supported](https://laravel.com/docs/11.x/eloquent#composite-primary-keys) by Eloquent models.

You have to add the `id` incrementing column to the pivot table.

**Reply by cvassios-eibach** [reply]
Thank you! @karandatwani92
